db/columns/utils.py

```diff
@@ -34,4 +35,12 @@ def get_enriched_column_table(table, engine=None):
 def init_mathesar_table_column_list_with_defaults(column_list):
     default_columns = get_default_mathesar_column_list()
     given_columns = [MathesarColumn.from_column(c) for c in column_list]
+    given_column_names = [col_name.name for col_name in given_columns if col_name.name == constants.ID]
+    if len(given_column_names) > 0:
+        default_columns = [column for column in default_columns if column.name not in given_column_names]
+        for c in given_columns:
+            if c.name == constants.ID:
+                c.type = Integer()
+                c.primary_key = True
+            c.autoincrement = False
     return default_columns + given_columns
```

Comment:
I'm sorry, my previous comment wasn't clear.  My point is, in this section, we should use the default `constants.ID` column, but modify it with `autoincrement=False` .  The reason for that is that this will make sure that the resulting `ID` column will always be defined by the default `ID` column, with only the `autoincrement=False` change.  This makes the code base easier to maintain, since otherwise the `ID` column is defined in 2 places.  Try something like:

```suggestion
    default_columns = get_default_mathesar_column_list()
    given_columns = [MathesarColumn.from_column(c) for c in column_list if c.name != constants.ID]
    if len(given_columns) != len(column_list):  # if ID column given
        for col in default_columns:
            if col.name == constants.ID:
                col.autoincrement = False

    return default_columns + given_columns
```